src/pal/src/CMakeLists.txt

```diff
@@ -161,16 +187,29 @@ set(SOURCES
 add_library(coreclrpal
   STATIC
   ${SOURCES}
+  ${ARCH_SOURCES}
   ${PLATFORM_SOURCES}
 )
 
 if(CMAKE_SYSTEM_NAME STREQUAL Linux)
-  target_link_libraries(coreclrpal
-    pthread
-    rt
-    dl
-    unwind
-  )
+  # On ARM linking with libunwind will break C++ exceptions unless we first
+  # link with gcc_s, this is a libunwind issue
+  if(PAL_CMAKE_PLATFORM_ARCH_ARM)
+    target_link_libraries(coreclrpal
+      gcc_s
+      pthread
+      rt
+      dl
+      unwind
+    )
+  else()
+    target_link_libraries(coreclrpal
+      pthread
+      rt
+      dl
+      unwind
+    )
```

Comment:
Also it seems like we can unify this to not duplicate pthread, rt, dl, and unwind, and just prepend in the arm case.